src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs

```diff
@@ -204,11 +153,24 @@ private static Version ToVersion(string versionString)
             }
         }
 
-        private static DistroInfo GetDistroInfo() => new DistroInfo()
+        private static DistroInfo GetDistroInfo()
         {
-            Id = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystem,
-            VersionId = ToVersion(Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystemVersion)
-        };
+            DistroInfo result = new DistroInfo();
+
+            foreach (string line in File.ReadAllLines("/etc/os-release"))
```

Comment:
are we sure the file `/etc/os-release` always exist? I mean do we need to check if it exist before reading it?

Reply:
My thinking was that if we get here and that file doesn't exist, we should find out about it (i.e. throw an exception and fail the test). Because if the file doesn't exist, I'm not sure what to put into this struct.

Two of the places that call this all check if we are running on Linux. Those should be fine, as I believe `/etc/os-release` exists on all Linuxes.

The one other place that calls this falls through `IsWindows`, and `IsOSX` checks, so maybe that one might be problematic on FreeBSD.

Reply:
I would say tolerate the failure of the id/version check and return some basic values (like id = "" and Version(0)) this should make the test that depends on these properties will still fail as the code will behave differently than expected. I am not feeling strong about it anyway so, I'll leave it to you to decide and I am ok either way we go as long as we'll not mistakenly fail in legitimate distros.